lua/ttt2/libraries/modelbuilder.lua

```diff
@@ -0,0 +1,316 @@
+---
+-- Used to build custom world and view models. 
+-- @author Mineotopia
+-- @module modelbuilder
+
+---@class ModelData
+---@field type string The type of this model, it can be "Model", "SPrite" or "Quad"
+---@field model string The path to the model file
+---@field bone string The name of the reference bone
+---@field rel string The name of a related identifier
+---@field pos Vector The position offset of the model
+---@field angle Angle The rotation offset of the model
+---@field size Vector The size multiplier of the model
+---@field color Color The color for the model
+---@field surpresslightning boolean Set to true if the engine lightning should be suppressed
+---@field material string The model material
+---@field skin number The skin index of the model
+---@field bodygroup table A table of bodygroups
+
+---@class BoneData
+---@field scale Vector The scale multiplier
+---@field pos Vector The position offset
+---@field angle Angle The rotation offset
+
+if SERVER then
+	AddCSLuaFile()
+
+	return
+end
+
+modelbuilder = {}
+
+local propertiesMaterial = {"nocull", "additive", "vertexalpha", "vertexcolor", "ignorez"}
+
+---
+-- Creates a clientside model that is used as a view or world model with the provided model data.
+-- @param Entity wep The weapon for which the model should be created
+-- @param ModelData modelData The model data for the model
+-- @realm client
+function modelbuilder.CreateModel(wep, modelData)
+	-- handle a model being added to the view or world model
+	if modelData.type == "Model" and modelData.model and modelData.model ~= ""
+		and (not IsValid(modelData.modelEnt) or modelData.createdModel ~= modelData.model)
+		and string.find(modelData.model, ".mdl") and file.Exists (modelData.model, "GAME")
+	then
+		modelData.modelEnt = ClientsideModel(modelData.model, RENDER_GROUP_VIEW_MODEL_OPAQUE) --todo check correct render group
```

Comment:
Check the todo and remove afterwards

Reply:
I did, just forgot to remove it. The thing is, it works and I took this line from the old addon. HOWEVER this render group doesn't exist in the docs. So I'm not sure here

Reply:
If it works, maybe add a comment, that it is known that this is unknown. Just to prevent someone from changing it just because it does not exist in the docs